portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -75,16 +89,65 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
-    /** Use query to populate a list of Comment objects. */
+    /* Use query to populate a list of Comment objects. */
     ArrayList<Comment> comments = new ArrayList<Comment>();
-    
-    for (Entity entity : commentResults.asIterable(options)) {
-        long id = entity.getKey().getId();
-        String message = (String) entity.getProperty("message");
-        long timestamp = (long) entity.getProperty("timestamp");
+    ArrayList<Entity> commentEntities = new ArrayList<Entity>();
+
+    /* Add entities to a list for convenient access later */
+    for (Entity entity: commentResults.asIterable(options)) {
+      commentEntities.add(entity);
+    }
+
+    /* Create a threadpool to perform multiple translations in parallel */
+    ExecutorService executor = Executors.newFixedThreadPool(10);
+    /* A list to hold pending translations in the order they were added. */
+    CompletableFuture<String>[] translationFutures = new CompletableFuture[commentEntities.size()];
+
+    /* Iterate entities and start a translation async for each */
+    for (int i = 0; i < commentEntities.size(); i++) {
+      Entity entity = commentEntities.get(i);
+      
+      /* Lambda expression that captures the entity, performs translation, returns translated message. */
+      Supplier<String> translationTask = () -> {
+        String message = (String) entity.getProperty("message"); 
+        
+        /* A new translation service is created each time because Translate may not be threadsafe.*/
+        Translate translate = TranslateOptions.getDefaultInstance().getService();
+        Translation translation = translate.translate(
+          message,
+          Translate.TranslateOption.targetLanguage(languageCode == null ? "en" : languageCode)
+        );
+        return translation.getTranslatedText();
+      };
+
+      translationFutures[i] = CompletableFuture.supplyAsync(translationTask, executor);
+    }
 
-        Comment comment = new Comment(id, message, projectId, timestamp);
-        comments.add(comment);
+    /* Block until all translations are complete.
+     * This isn't ideal but I'm willing to bet traffic isn't a huge problem. 
+     * For this site, fast load for one >>> handle many users.
+     */
+    CompletableFuture.allOf(translationFutures).join();
+
+    /* Add translated queries for response */
+    for (int i = 0; i < commentEntities.size(); i++) {
+      Entity entity = commentEntities.get(i);
+      long id = entity.getKey().getId();
+      long timestamp = (long) entity.getProperty("timestamp");
+      String email = (String) entity.getProperty("email");
+
+      String translatedMessage = "";
+      try {
+        translatedMessage = translationFutures[i].get();
+      } catch (Exception unexpected) {
+        /* Really the only thing that can cause an error here is 
+         * .get() being called before the task is complete.
+         * That is taken care of by the .join() block above.
+         */
```

Comment:
Generally, swallowing exceptions silently is a bad practice unless the exception is truly meaningless.  I'd suggest logging an error as a minimum so that if a user reports that the site is behaving strangely while translating, there'll be an indication if somehow exceptions were being thrown here.